rofl-scheduler/src/types.rs

```diff
@@ -39,3 +42,35 @@ pub struct TerminateRequest {
     /// Whether the storage should be wiped.
     pub wipe_storage: bool,
 }
+
+/// Name of the metadata key used to store the delegated permissions.
+pub const METADATA_KEY_PERMISSIONS: &str = "net.oasis.scheduler.permissions";
+
+/// Name of the action for viewing machine logs.
+pub const ACTION_LOG_VIEW: &str = "log.view";
+
+/// Instance permissions for different actions.
+pub type Permissions = BTreeMap<String, BTreeSet<Address>>;
```

Comment:
Should we also introduce an enum type for permission names? Or is there an usecase for these to be arbitrary strings so that rofl apps could also use it for application specific permissions?

If we support arbitrary strings, we should probably have some limits on the key sizes? Maybe also on the number of addresses in the value?

Reply:
It's fine then 👍